### ISSUE TITLE
New LaTeX version breaks jss.cls and ajs.cls

### ISSUE THREAD

**Issue by dsejas** [matthias-da/ajs-public#2]
**General comment:** I recently re-installed my LaTeX distribution and every time I try to compile with `ajs.cls`, I get an error message. After a whole afternoon trying to find a solution, I just found out that the latest LaTeX version breaks `jss.cls` and, therefore, also `ajs.cls`.

**Description of the problem:** The `ajs.cls` class produces the error message

    ! Undefined control sequence.
    \hyper@linkurl ...tionraw >>}\relax \Hy@colorlink 
                                                      \@urlcolor #1\Hy@xspace@en...
    l.64 \begin{document}

**How to reproduce:** Download the latest `ajs.cls` file and try to compile `guidelinesAJS.tex`. (Any file using `ajs.cls` will have this problem.)

**System information:** Kubuntu 20.04.2, TeXlive 2020 (installed on 2021-03-02), pdfTeX 3.14159265-2.6-1.40.21.

**Additional information:** The error message disappears as soon as all `\href` and `\url` commands are moved after the `\begin{document}`. This seems to be an issue with `\AtBeginDocument` hook, which has recently been changed. This affected the `jss.cls` class, and therefore, the `ajs.cls` class, which is based on it (see below).

**References:** [This issue report](https://github.com/rstudio/rticles/issues/329), specially [this comment](https://github.com/rstudio/rticles/issues/329#issuecomment-705713973) There is a [proposed solution](https://github.com/rstudio/rticles/pull/337/files).

**Comment by matthias-da** [comment]
Dear dsejas

sorry for the late reply cause by semester start at our university.

Thank you very much for your solution, this is really very much appreciated. 
And such community support as you did by solving this issue, keeps me going to run the journal without fees and 100% voluntary work. 

All the best
Matthias

**Comment by dsejas** [comment]
I am very happy to help! Thank you for your effort on this LaTeX format!

**Comment by zeileis** [comment]
Hi @matthias-da, I came across this issue today by coincidence and noticed that the solution you incorporated into your cls files was a preliminary (and somewhat brute-force) solution proposed by the `rticles` maintainers. Shortly thereafter I had proposed an alternative solution that works in both old and new versions of LaTeX:

https://github.com/rstudio/rticles/issues/329#issuecomment-711999725

This has been incorporated in jss.cls on jstatsoft.org and in the base R distribution, and also in `rticles`:

https://github.com/rstudio/rticles/pull/355/commits/3c9951c9e28a44a9c4ef26dd565ea2a329b8ad93

I would recommend that you also compare your cls against a recent version of jss.cls to check whether you want to follow some of the updates.

Bonus: The next jss.cls will use {xcolor} instead of {color} in order to facilitate using the {orcidlink} package as we have started adding ORCID links in the PDFs. Let me know if you want to do something similar...
